lib/portlayer/event/collector/vsphere/collector.go

```diff
@@ -159,7 +163,21 @@ func evented(ec *EventCollector, page []types.BaseEvent) {
 		return
 	}
 
+	if len(page) == 0 {
+		return
+	}
+
+	// skip events already seen
```

Comment:
@dougm agree - I had a similar thought on Friday but hadn't had time to look into it more.